### PR TITLE
Fast-forward main changes to 2.0 branch

### PR DESCRIPTION
**Pull request by mikeshng**
Signed-off-by: Mike Ng <ming@redhat.com>

# Description of the change(s):
* all changes from main branch

## Why do we need this PR:
*  we need the 2.0 branch to be updated with the main branch changes.

## Issue reference: 
* https://issues.redhat.com/browse/ACM-1447

## Test API/Unit - Success
```script
?   	github.com/stolostron/hypershift-addon-operator/cmd	[no test files]
ok  	github.com/stolostron/hypershift-addon-operator/pkg/agent	5.490s	coverage: 70.0% of statements
?   	github.com/stolostron/hypershift-addon-operator/pkg/manager	[no test files]
?   	github.com/stolostron/hypershift-addon-operator/pkg/util	[no test files]
```

### CODE REVIEW

File: COMPONENT_VERSION [hunk 1]
```diff
@@ -1 +1 @@
-2.5.0
\ No newline at end of file
+2.5.1
```
Comment (jnpacker) [inline COMPONENT_VERSION:1]: I think this should be 2.0.1

Comment (mikeshng) [reply COMPONENT_VERSION]: Done as suggested.

File: go.mod [hunk 2]
```diff
@@ -7,19 +7,19 @@ require (
 	github.com/go-logr/zapr v1.2.0
 	github.com/onsi/ginkgo/v2 v2.1.3
 	github.com/onsi/gomega v1.18.1
-	github.com/openshift/hypershift v0.0.0-20220511163932-ad143ad73524
-	github.com/openshift/library-go v0.0.0-20220112153822-ac82336bd076
-	github.com/spf13/cobra v1.2.1
+	github.com/openshift/hypershift v0.0.0-20220614212330-01bbfba6b681
```
Comment (zhujian7) [inline go.mod:10]: Please remember to update the version for downstream: https://gitlab.cee.redhat.com/mce-cpaas-midstream/hypershift/-/blob/multicluster-engine-2.0-rhel-8/upstream_sources.yaml#L53

Comment (mikeshng) [reply go.mod:10]: @rokej said he will help me with the DS update.

**Pull request merged**